crates/energy/src/battery.rs

```diff
@@ -0,0 +1,203 @@
+use bevy::prelude::*;
+
+pub(crate) struct BatteryPlugin;
+
+impl Plugin for BatteryPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_system(discharge_battery);
+    }
+}
+
+/// The rate at which the battery discharges in Joules per second.
+const DISCHARGE_RATE: f64 = 30_000.;
+/// The minimum energy level in kiloJoules at which the unit can still move.
+const MIN_MOVE_ENERGY: EnergyUnit = EnergyUnit::Kilojoules(74.); // based on how many a car uses per second
```

Comment:
Why not 0? I think that it should be allowed to get to 0. for the most basic unit activity, i.e. movement.